File: solutions/027.remove-element/remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
const removeElement = (nums, val) => {
  for (let i = 0; i < nums.length; i += 1) {
    if (nums[i] === val) {
      nums.splice(i, 1)
      i--
    }
  }
  return nums.length
};

File: solutions/027.remove-element/remove-element.py
class Solution:
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        ret = []

        for n in nums:
            if n != val:
                ret.append(n)

        for i in range(len(ret)):
            nums[i] = ret[i]

        return len(ret)
